sway-lsp/src/utils/debug.rs

```diff
@@ -39,7 +39,7 @@ where
     tokens
         .filter_map(|entry| {
             let (ident, token) = entry.pair();
-            if token.typed.is_some() {
+            if token.as_typed().is_some() {
                 Some(warning_from_ident(ident))
             } else {
                 None
```

Comment:
If it's not impossible because of data we might have thrown away by now, will we ever need to go the other way (even just during development)?

Reply:
we generally don't need to go back down the compiler chain once we get to typed tokens. We can always retrieve each stage, i.e lexed, parsed, and typed in vscode by using the command palette with is handy for development.